client/blocks/do-not-sell-dialog/use-prevent-scroll.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+
+/*
+    Pages in wp-calypso that have footer (and require Do Not Sell Dialog) are scrollable (body height > 100vh).
+    This hook prevents scrolling when the dialog is open while not implementing any new external dependencies for the dialog.
+
+    Solution reference: https://github.com/reactjs/react-modal/issues/191#issuecomment-1155467966
+*/
+export const usePreventScroll = ( isOpen: boolean ) => {
+	useEffect( () => {
+		if ( isOpen ) {
+			const x = window.scrollX;
+			const y = window.scrollY;
+
+			const disableScroll = () => {
+				window.scrollTo( x, y );
+			};
+			window.addEventListener( 'scroll', disableScroll );
+			return () => {
+				window.removeEventListener( 'scroll', disableScroll );
+			};
```

Comment:
As I understand, the [cleanup function runs on every prop change](https://reactjs.org/docs/hooks-effect.html#explanation-why-effects-run-on-each-update), so that hook should work correctly. But I agree, in general, it's better to keep the hook definition on one level, rather than adding scopes, to avoid confusion.

I'm removing this hook anyway for now

Reply:
Just to clarify: the cleanup function should work on every prop change, but it didn't in this case, because we were returning the cleanup function only if `isOpen` is truthy.